ViewControllers/DetailViewController.swift

```diff
@@ -202,40 +238,67 @@ class DetailViewController: UIViewController {
 
             imageView.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 16),
             imageView.centerXAnchor.constraint(equalTo: titleLabel.centerXAnchor),
-            imageView.widthAnchor.constraint(equalTo: view.widthAnchor, multiplier: 0.61),
+            imageView.widthAnchor.constraint(equalTo: contentView.widthAnchor, multiplier: 0.61),
             imageView.heightAnchor.constraint(equalTo: imageView.widthAnchor, multiplier: 1.41),
             
-            releaseStack.leftAnchor.constraint(equalTo: view.layoutMarginsGuide.leftAnchor),
+            releaseStack.leftAnchor.constraint(equalTo: contentView.layoutMarginsGuide.leftAnchor),
             releaseStack.topAnchor.constraint(equalTo: imageView.bottomAnchor, constant: 35),
 
             durationStack.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
             durationStack.topAnchor.constraint(equalTo: releaseStack.bottomAnchor, constant: 8),
 
             movieDescriptionLabel.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
-            movieDescriptionLabel.rightAnchor.constraint(equalTo: view.layoutMarginsGuide.rightAnchor),
+            movieDescriptionLabel.rightAnchor.constraint(equalTo: contentView.layoutMarginsGuide.rightAnchor),
             movieDescriptionLabel.topAnchor.constraint(equalTo: durationStack.bottomAnchor, constant: 16),
 
             collectionView.leftAnchor.constraint(equalTo: releaseStack.leftAnchor),
             collectionView.topAnchor.constraint(equalTo: movieDescriptionLabel.bottomAnchor, constant: 16),
             collectionView.rightAnchor.constraint(equalTo: movieDescriptionLabel.rightAnchor),
-            collectionView.heightAnchor.constraint(equalToConstant: 200),
+            collectionView.heightAnchor.constraint(equalToConstant: calculateCollectionViewHeight())
         ])
     }
     
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        scrollView.contentSize = CGSize(width: view.frame.size.width, height: calculateScrollViewContentSize())
+    }
+    
+    private func calculateCollectionViewHeight()-> CGFloat {
+        var languageRows = languages.count / 4
+        languages.count % 4 != 0 ? languageRows += 1 : nil
+        let languageRowsHeight = (languageRows * 27) + ((languageRows - 1) * 4)
+        
+        var genreRows = genres.count / 4
+        genres.count % 4 != 0 ? genreRows += 1 : nil
+        let genreRowsHeight = (genreRows * 27) + ((genreRows - 1) * 4)
+        
+        return CGFloat(languageRowsHeight + 16 + genreRowsHeight)
+    }
+    
+    private func calculateScrollViewContentSize()-> CGFloat {
+        let lastItemOrigin = collectionView.frame.origin.y
+        let lastItemHeight = collectionView.frame.size.height
+        
+        print(lastItemHeight)
+        
+        let scrollViewContentSize = lastItemHeight + lastItemOrigin
+        return scrollViewContentSize
+    }
```

Comment:
It's fine manually calculating the collection content size, but we can use UIKit itself to tell us the size and leave the work of calculating it to autolayout itself.
```swift
collectionView.setNeedsLayout() // invalidates current layout and triggers a layout update in the next update cycle
collectionView.layoutIfNeeded() // updates the layout immediately if needed
let updatedContentSize = collectionView.contentSize
```